src/state/user/reducer.ts

```diff
@@ -210,5 +222,32 @@ export default createReducer(initialState, builder =>
     })
     .addCase(toggleTopTrendingTokens, state => {
       state.showTopTrendingSoonTokens = !state.showTopTrendingSoonTokens
+    })
+    .addCase(toggleFavoriteToken, (state, { payload: { chainId, isNative, address } }) => {
+      let favoriteTokens = state.favoriteTokensByChainId[chainId]
+      if (!favoriteTokens) {
+        favoriteTokens = {
+          includeNativeToken: false,
+          addresses: [],
+        }
+        state.favoriteTokensByChainId[chainId] = favoriteTokens
+        return
+      }
+
+      if (isNative) {
+        const previousValue = favoriteTokens.includeNativeToken
+        favoriteTokens.includeNativeToken = !previousValue
+        return
+      }
+
+      if (address) {
+        // this is intentionally added, to remove compiler error
+        const index = favoriteTokens.addresses.findIndex(addr => addr === address)
+        if (index === -1) {
```

Comment:
use `favoriteTokens.addresses.some(addr => addr === address)`

Reply:
I need the index in line 250

Reply:
ah, ok